uwblueprint/childrens-aid-society
# Intake footer

## Notion ticket link

[Create intake form footer navigation component](https://www.notion.so/uwblueprintexecs/134e1f159caa4f73939d11a700c325ad?v=39cf7d5a98384eb9af2bff2265d0ac8a&p=3e8a3cb2f18742fa91e7f69326a78bdc&pm=s)


## Implementation description
* Create footer component which is places in each individual registration page. 
* Footer handles navigation and saving state on click
* Add header stepper to each individual page
* Stepper handle navigation and saving state on click



## Steps to test
1. Edit fields and navigate via either stepper or footer. Notice that the fields (except for Individual details) page is being updated.
[Untitled_ Feb 4, 2023 3_44 AM.webm](https://user-images.githubusercontent.com/117593511/216758018-2a22fa2a-dd64-4031-9d58-b29642c89d5d.webm)



## What should reviewers focus on?
* I didn't handle saving state for individual details because I felt the implementation was rather involved and out of the scope of this PR 
* I noticed that Eugene had worked on some button features for hiding buttons on read only. I remove those code bits and just added buttons via the footer
* The need to have a previous button within the footer feels rather important to me. Especially int he case of the the review page where a header stepper is not provided, I feel that the UI would look clearer having the previous button in the footer and not in the nested page component. 
* I didn't implement the clear page button feature. This was because the previous [PR being built on-top of](https://github.com/uwblueprint/childrens-aid-society/pull/95) didn't mention such requirement.
* Since I had to put the header stepper inside of the registration page, the stepper inherited some additional css properties which reduced the width to the width of the page element. I couldn't find a clean way to increase the width back to normal.

## Checklist
